### DISCUSSION TITLE
Hide Help in edit mode

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [NemesisRE/kiosk-mode#119] [Ideas]
Not sure if we discussed this before, but here goes....

The final menu bar item I hide using card_mod, is the help icon in  edit mode:

<img width="371" alt="Scherm­afbeelding 2023-10-04 om 10 54 55" src="https://github.com/NemesisRE/kiosk-mode/assets/33354141/b7e5edc6-f8f6-4c4a-9ee1-3128292d95e9">

toggle the boolean:

<img width="500" alt="Scherm­afbeelding 2023-10-04 om 10 55 12" src="https://github.com/NemesisRE/kiosk-mode/assets/33354141/c71c968d-eee5-464a-b2f6-d91ec1aa23b8">

and the result is:
<img width="369" alt="Scherm­afbeelding 2023-10-04 om 10 55 24" src="https://github.com/NemesisRE/kiosk-mode/assets/33354141/712b1fb1-ea3b-4779-a0e5-8d2c9bd18f78">

its a bit ugly now because of the template in the header (which I also enable using a toggle, might be too specific for kioskmode? )

but its done using this mod:

```
  card-mod-root-yaml: |
    .: |
      /* This hides the help button when in edit mode*/
      a.menu-link[target="_blank"] {
        {% if is_state('input_boolean.hide_help','on') %}display: none;{% endif %}
      }
```

maybe this is something you could consider for Kiosk-mode and do better....?

btw, did you ever consider a toggle for moving the header to the footer? We can do that via card_mod, but it requires more than a single line in mods, so I suppose to would be appreciated by the community if Kiosk-mode could take care of that too.?

**Comment by elchininet** [comment]
Hey @Mariusthvdb,

What if you do something like this?:

```yaml
card-mod-root-yaml: |
    .: |
      /* This hides the help button when in edit mode*/
      a.menu-link[target="_blank"] {
        {% if is_state('input_boolean.hide_help','on') %}visibility: hidden;{% endif %}
      }
```

**Reply by Mariusthvdb** [reply]
ooh, thats much better yes!
<img width="364" alt="Scherm­afbeelding 2023-10-04 om 13 27 34" src="https://github.com/NemesisRE/kiosk-mode/assets/33354141/1ab16d2a-7485-4948-a077-d889e74ea8da">

nice

**Reply by elchininet** [reply]
You can also add some negative margin to the element so the text gets closer to the `Klaar` button.

**Reply by Mariusthvdb** [reply]
how would we go about add that margin? ive always believed we could only hide/obfuscate the existing elements, not change its position there?

**Reply by elchininet** [reply]
Try this:

```yaml
card-mod-root-yaml: |
    .: |
      /* This hides the help button when in edit mode*/
      a.menu-link[target="_blank"] {
        {% if is_state('input_boolean.hide_help','on') %}visibility: hidden;margin-right: -20px;{% endif %}
      }
```

Decrease the margin to get the elements closer.

**Reply by Mariusthvdb** [reply]
thank you, that is very nice indeed.
Ive added this to my card-mod-them, also in the hist:
```
      /* This hides the help button when in edit mode*/
      a.menu-link[target="_blank"] {
        {% if is_state('input_boolean.hide_edit_mode_help','on') %}
        visibility: hidden;
        margin-right: -30px;
        {% endif %}
      }
```

**Comment by elchininet** [comment]
>btw, did you ever consider a toggle for moving the header to the footer? We can do that via card_mod, but it requires more than a single line in mods, so I suppose to would be appreciated by the community if Kiosk-mode could take care of that too.?

That was one of the functionalities of [custom-header](https://github.com/maykar/custom-header). I don‘t know if there are some forks alive nowadays, but it should not be included into `kiosk-mode`. Anything that falls outside of avoiding non-authorised users to interact with sensitive information is outside of the scope, mainly those that are a style change or a restructure of the layout.

**Reply by elchininet** [reply]
It would be good to revive that project, it had a lot of users that found it helpful. Let me know if you find some forks alive, if not that would be a good project to revive.

**Reply by Mariusthvdb** [reply]
agreed 100%
I believe I was one of the last one standing using it, until it really was borked by core HA.....

I use my card_mod_themes to do all of that stuff now, https://gist.github.com/Mariusthvdb/9069c8c4dcad6324035a679244bbfdf1 and it has always felt less robust than custom-header.
Dont believe any forks are alive, so yeah, if you would feel like it.... Id sure applaud that, and would be willing to test run anything.

**Reply by elchininet** [reply]
Whoa, that is a huge theme-mode 😅
Perfect, when I find some time, I‘ll try to work on it.
